### PR TITLE
Data Access View tags

### PR DESCRIPTION
**Pull request by arcturus2**
### Description of the Change

:art: Added tags for Data Access View plugins
:white_check_mark: Added unit tests for new plugins that can now be
tested easily

I have omitted the `DataAccessPane` as I have a different PR working through that so to avoid serious merge conflicts I will update any pending tags in that PR.

I have also added unit tests for the plugins that where taken out from being inline.

### CODE REVIEW

File: CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/experimental/TestParametersPlugin.java [hunk 1]
```diff
@@ -107,6 +109,7 @@
     @ServiceProvider(service = Plugin.class)
 })
 @Messages("TestParametersPlugin=Test Parameters")
+@PluginInfo(pluginType = PluginType.SELECTION, tags = {"EXPERIMENTAL", "SELECT"})
 public class TestParametersPlugin extends RecordStoreQueryPlugin implements DataAccessPlugin {
```
Comment (antares1470) [inline CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/experimental/TestParametersPlugin.java:113]: Given it sits in the Developer section of DAV, would it make sense to tag with DEVELOPER? Tagging with SELECT also is a bit confusing to me

File: CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/templates/RecordStoreDropper.java [hunk 2]
```diff
@@ -111,4 +100,31 @@ protected RecordStore query(final RecordStore query, final PluginInteraction int
         return null;
     }
 
+    @PluginInfo(pluginType = PluginType.IMPORT, tags = {"IMPORT"})
+    public static class RecordStoreDropperRecordStoreQueryPlugin extends RecordStoreQueryPlugin {
```
Comment (antares1470) [inline CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/templates/RecordStoreDropper.java:104]: the name is a bit confusing to me as it sounds more like a plugin type than a plugin. The previous name included a mention of record store to graph. Could be good to include that somewhere here (unless of course that was wrong)?

File: CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/templates/TSVDropper.java [hunk 3]
```diff
@@ -144,4 +133,33 @@ protected RecordStore query(final RecordStore query, final PluginInteraction int
 
         return null;
     }
+
+    @PluginInfo(pluginType = PluginType.IMPORT, tags = {"IMPORT"})
+    public static class TSVDropperRecordStoreQueryPlugin extends RecordStoreQueryPlugin {
```
Comment (antares1470) [inline CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/templates/TSVDropper.java:138]: similar

File: CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/templates/RecordStoreDropper.java [hunk 4]
```diff
@@ -111,4 +100,31 @@ protected RecordStore query(final RecordStore query, final PluginInteraction int
...
+
+        private final RecordStore recordStore;
+
+        public RecordStoreDropperRecordStoreQueryPlugin(final RecordStore recordStore) {
+            this.recordStore = recordStore;
+        }
+
+        @Override
+        protected RecordStore query(RecordStore query, PluginInteraction interaction, PluginParameters parameters) throws InterruptedException, PluginException {
```
Comment (antares1470) [inline CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/templates/RecordStoreDropper.java:113]: finals

File: CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/templates/TSVDropper.java [hunk 5]
```diff
@@ -144,4 +133,33 @@ protected RecordStore query(final RecordStore query, final PluginInteraction int
...
+        private final RecordStore recordStore;
+        private final List<File> files;
+
+        public TSVDropperRecordStoreQueryPlugin(final RecordStore recordStore, final List<File> files) {
+            this.recordStore = recordStore;
+            this.files = files;
+        }
+
+        @Override
+        protected RecordStore query(RecordStore query, PluginInteraction interaction, PluginParameters parameters) throws InterruptedException, PluginException {
```
Comment (antares1470) [inline CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/templates/TSVDropper.java:149]: finals

**Review by antares1470 — CHANGES_REQUESTED** [review]

File: CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/experimental/TestParametersPlugin.java (hunk 1 above)
Comment (arcturus2) [reply CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/experimental/TestParametersPlugin.java:113]: I'm not sure how or why `SELECT` came into this. I am certain I was meant to put `MODIFY`. I'll have another go and thank you for picking up on this.

Comment (arcturus2) [reply CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/experimental/TestParametersPlugin.java]: Fixed

File: CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/templates/RecordStoreDropper.java (hunk 2 above)
Comment (arcturus2) [reply CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/templates/RecordStoreDropper.java]: Thanks and I agree. I updated the name based on what the previous name was and called it `RecordStoreDropperToGraphPlugin`

File: CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/templates/TSVDropper.java (hunk 3 above)
Comment (arcturus2) [reply CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/templates/TSVDropper.java]: Updated it to `TSVDropperToGraphPlugin`

File: CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/templates/RecordStoreDropper.java (hunk 4 above)
Comment (arcturus2) [reply CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/templates/RecordStoreDropper.java]: Fixed

File: CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/templates/TSVDropper.java (hunk 5 above)
Comment (arcturus2) [reply CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/templates/TSVDropper.java]: Fixed

**Review by antares1470 — APPROVED** [review]

File: CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/experimental/TestParametersPlugin.java [hunk 6]
```diff
@@ -107,6 +109,7 @@
     @ServiceProvider(service = Plugin.class)
 })
 @Messages("TestParametersPlugin=Test Parameters")
+@PluginInfo(pluginType = PluginType.SELECTION, tags = {"DEVELOPER", "MODIFY"})
```
Comment (aldebaran30701) [inline CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/experimental/TestParametersPlugin.java:112]: plugintype may be better suited to UPDATE?

**Review by aldebaran30701 — APPROVED** [review]
Just a minor note

File: CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/experimental/TestParametersPlugin.java (hunk 6 above)
Comment (arcturus2) [reply CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/experimental/TestParametersPlugin.java]: thank you @aldebaran30701, fixed!

**Pull request merged**